File: Difficulty: Medium/Rotate Array/rotate-array.cpp
class Solution {
  public:

    // Function to rotate an array by d elements in counter-clockwise direction.
    void rotate(vector<int> &a, int lo, int hi){
        while (lo<hi){
            swap(a[lo], a[hi]);
            lo++, hi--;
        }
    }
    void rotateArr(vector<int>& arr, int d) {
        // code here
        int n = arr.size();
        d = d%n;
        rotate(arr, 0, d-1);
        rotate(arr, d, n-1);
        rotate(arr, 0, n-1);
    }
};

File: Difficulty: Medium/Rotate Array/rotate-array.java
class Solution {
	static void rotateArr(int arr[], int d) {
		int n = arr.length;	
		d = d % n;
		rotate(0, d - 1, arr);
		rotate(d, n - 1, arr);
		rotate(0, n - 1, arr);
	}
	
	static void rotate(int i, int j, int[] a) {
		while (i<j) {
			int temp = a[i];
			a[i] = a[j];
			a[j] = temp;
			i++;
			j--;
		}
	}
}
